code/datums/emergency_calls/beacon-reinforcements/freelancers.dm

```diff
@@ -0,0 +1,47 @@
+/datum/emergency_call/freelancer/beacon
+	name = "Freelancer  Infantry (Squad) (Beacon)"
+	mob_max = 8
+	max_medics = 1
+	mob_min = 1
+	spawn_max_amount = TRUE
+
+/datum/emergency_call/freelancer/beacon/New()
+	. = ..()
+	arrival_message = "[MAIN_SHIP_NAME], this is the Free Vessel Nellie. We'll take your contract for aid, coming soon."
+	objectives = "Render assistance towards the USCM. You've already been paid by the ship's automated systems.."
+
+/datum/emergency_call/freelancer/beacon/print_backstory(mob/living/carbon/human/spawning_mob)
+	to_chat(spawning_mob, SPAN_BOLD("You started off in the Neroid Sector as a colonist seeking work at one of the established colonies."))
+	to_chat(spawning_mob, SPAN_BOLD("The withdrawl of United American forces in the early 2180s, the system fell into disarray."))
+	to_chat(spawning_mob, SPAN_BOLD("Taking up arms as a mercenary, the Freelancers have become a powerful force of order in the system."))
+	to_chat(spawning_mob, SPAN_BOLD("While they are motivated primarily by money, many colonists see the Freelancers as the main forces of order in the Neroid Sector."))
+	to_chat(spawning_mob, SPAN_BOLD("Ensure they are not destroyed."))
+
+
+/datum/emergency_call/freelancer/beacon/create_member(datum/mind/spawning_mind, turf/override_spawn_loc)
+	var/turf/spawn_loc = override_spawn_loc ? override_spawn_loc : get_spawn_point()
+
+	if(!istype(spawn_loc))
+		return //Didn't find a useable spawn point.
+
+	var/mob/living/carbon/human/mob = new(spawn_loc)
+	spawning_mind.transfer_to(mob, TRUE)
+
+	if(!leader && HAS_FLAG(mob.client.prefs.toggles_ert, PLAY_LEADER) && check_timelock(mob.client, JOB_SQUAD_LEADER, time_required_for_job && (spawning_mind)))
```

Comment:
```suggestion
	if(!leader && (!spawning_mind || (HAS_FLAG(mob.client.prefs.toggles_ert, PLAY_LEADER) && check_timelock(mob.client, JOB_SQUAD_LEADER, time_required_for_job))))
```
and implement this for all the similar occurrences. Basically if you don't have a mind (a client willing to roll the role), you have no one to check for preferences.

Reply:
changed